Assignment invitation URL improvements

EricPickup opened the pull request:
Closes #2115.

This PR makes the assignment invitation URL stand out more on the assignment view page, and adds the ability to enable/disable the invitation URL through ajax calls directly on the page.

![6IImxsJjGk](https://user-images.githubusercontent.com/1490434/62907865-29425580-bd2a-11e9-9aed-49b02df97434.gif)

d12 commented:
Curious what the reasoning is behind adding so much whitespace around the invitation link. It's important, but a teacher typically copies it once and after that, they only care about the list of submissions. This work pushes the useful information _way_ down the page. Is there a better way we can design this? Possibly making better use of horizontal space that we aren't using right now?

d12 commented:
Gotcha :) I 100% agree that the checkbox lives on this page, that makes sense.

To show why I think the vertical space could be an issue, I pulled down the branch and un-fullscreened my Chrome (I don't know how many people use fullscreen apps vs non fullscreen, but I've seen both)

<img width="1679" alt="Screen Shot 2019-08-12 at 11 32 07 AM" src="https://user-images.githubusercontent.com/30920216/62877443-106b7d00-bcf5-11e9-89d2-cfa92f63530c.png">

Notice how headers, more headers, and the invitation link dominate the screen. This screenshot is also representative of users with resolutions below 1920*1080, or users who increase the font size in their browser.

With a bit of re-organization, I wonder if the invitation link could go within the assignment header where the download + settings buttons are.

I don't want to block any work, just wanted to give my two cents :)

stephaniegiang commented:
@femmebot @d12 I have a PR in the works that will make the changes that Brandon proposed in https://github.com/education/classroom/pull/1835 so this area will be moved to the header. Maybe that will help with the poor use of space.

EricPickup commented on app/views/shared/_invitation.html.erb:
```diff
@@ -1,9 +1,16 @@
-    <%= render partial: "shared/invitations/desktop", locals: local_assigns %>
-  <% end %>
-</form>
+<dt class=<%= mobile? ? "mobile-invitation-content" : "desktop-invitation-content" %>>
```
It looks like we have some flaws with this. The mobile/desktop class is decided on the page load (as seen on this line) so switching on inspect element with mobile will not work instantly and requires refresh. We could fix this in another issue using media queries.

This is what it looks like for me on mobile:
![image](https://user-images.githubusercontent.com/1490434/62907381-c64fbf00-bd27-11e9-94f5-749364acea77.png)

EricPickup commented on app/views/shared/_invitation.html.erb:
```diff
@@ -1,9 +1,16 @@
-  <% end %>
-</form>
+<dt class=<%= mobile? ? "mobile-invitation-content" : "desktop-invitation-content" %>>
+  <%= check_box_tag 'invitations_enabled',true, (@assignment.nil? ? @group_assignment.invitations_enabled : @assignment.invitations_enabled),
```
@femmebot do you mean to remove it only from creation or from the assignment settings page as well?
@d12 what are your opinions on this? I think it should remain in the settings page since it is still a setting. Would be weird keeping one specific setting outside of the settings page.

d12 replied:
I'll defer to @femmebot or the engineering team here to make the correct decision :)

EricPickup commented:
![image](https://user-images.githubusercontent.com/1490434/62907762-a3bea580-bd29-11e9-941c-16a88a6e5b11.png)

How does this spacing look? I removed all of the extra padding that was originally added. Also fixed the issue where the spinning octocat gif bumped the block up/down.

d12 commented:
This spacing looks much better, thanks!

EricPickup commented:
@femmebot let me know what you think 😄

JessRudder commented on app/assets/stylesheets/layout.scss:
```diff
@@ -432,6 +429,11 @@ hr {
   }
 }
 
+.desktop-invitation-content {
```
Open question (that I do not know the answer to) - Do we need the custom CSS or could this be achieved by conditionally adding/removing Primer classes?

EricPickup replied:
We do have a primer class for auto margin but I can't find anything for 50% width so we'd still have to add this class 😞I'll leave it as-if for now since we need the class either way and it's easier to distinguish the difference between mobile vs desktop in separate classes
